src/tracks/playabletrack/wavetrack/ui/SpectrumCache.cpp

```diff
@@ -171,17 +167,12 @@ bool SpecCache::CalculateOneSpectrum
          }
 
          if (myLen > 0) {
+            constexpr auto iChannel = 0u;
+            mSampleCacheHolder.emplace(clip.GetSampleView(iChannel, from, myLen));
             floats.resize(myLen);
-            auto data = floats.data();
-            constexpr auto backwards = false;
-            if (sequence.GetFloats(
-                   0u, 1u, &data,
-                   sampleCount(floor(0.5 + from.as_double() + offset * rate)),
-                   myLen, backwards, FillFormat::fillZero,
-                   // Don't throw in this drawing operation
-                   false))
```

Comment:
Notice how `GetFloats` had a `mayThrow` argument defaulting true but passed false here.

I added that years ago.  I reasoned that if reading sample blocks failed for any reason, but only in display code, it should be non-throwing, and instead just zeroes should be substituted.

But now that is lost with this transformation.

See where `SampleBlock.cpp` used to use the `mayThrow` argument.

But that was a guard around the concrete class that implemented `DoGetSamples`.

So I think `SqliteSampleBlock::GetFloatSampleView` should now add a try-catch around its new call to `DoGetSamples` and the catch block needs to add zeroes.

So that the behavior in exceptional cases does NOT change.

Reply:
At last I know what the `mayThrow` is about - my bad not for having actively searched its meaning.
`GetFloatSampleView` is used for both display and playback, though, so adding a try-catch would override the current playback, throwing behaviour.
It's not too much work to add this boolean to `GetFloatSampleView`. Otherwise we'd have to choose one behaviour for both use cases - always catching would also affect playback behaviour.
Added a commit that does this.